ui/analyse/src/serverSideUnderboard.ts

```diff
@@ -30,7 +30,23 @@ export default function (element: HTMLElement, ctrl: AnalyseCtrl) {
     unselect = (chart: Highcharts.ChartObject) => chart.getSelectedPoints().forEach(point => point.select(false));
   let lastInputHash: string;
 
+  const updateGifLinks = (fen: Fen) => {
+    positionGifLink.href = xhrUrl(`/export/gif/${fen.replace(/ /g, '_')}`, {
+      color: ctrl.bottomColor(),
+      lastMove: ctrl.node.uci,
+      variant: ctrl.data.game.variant.key,
+      theme: document.body.dataset.boardTheme,
```

Comment:
On the other hand results depending on the session cannot be cached and delivered via the CDN.

Reply:
Yeah, this is an interesting tradeoff to consider.

1. Frontend supplies theme/piece to lila backend
2. Lila backend supplies theme/piece before requesting GIF from lila-gif

I went with (1) so that all of our GIF export endpoints were "pure" in the sense that they didn't rely on session info. I actually feared that browser caching would interfere with (2). If you generate a gif at e.g. https://lichess1.org/game/export/gif/CSguMKFd.gif, whatever session generated that gif will determine the theme. If the user changes their theme and regenerates the same gif, I imagine that the gif won't change based on the browser's cache or CDN cache. Also if someone else generates a gif at the same URL, will they receive what they expect?

That all felt like a dealbreaker for (2), but the downside is definitely scattering a lot of theme/piece piping throughout the codebase.

Reply:
Maybe a `302 Found` redirect from `https://lichess.org/...` (from new routes) using the session state to a pure `https://lichess1.org/...` URL?

Reply:
So I'm noticing that for generating a new gif, the CDN URL `lichess1.org` `301` redirects back to `lichess.org`, presumably because it is a cache miss. Maybe there is a lag time there before it is cached properly, because upon repeatedly hitting that same CDN URL it continues to 301. My worry is that by having `lichess.org` redirect to `lichess1.org`, we could create an infinite loop 🤔

Reply:
That's strange/unexpected. What's the exact redirect that you're seeing?

Reply:
https://lichess1.org/export/gif/8/pp1rqpk1/3r2p1/P1pBp3/2P1PR1p/1P5P/1Q3PPK/R7_b_-_-_0_29?color=white&lastMove=f3f4&variant=standard
seems to be consistently redirecting to
https://lichess.org/export/gif/8/pp1rqpk1/3r2p1/P1pBp3/2P1PR1p/1P5P/1Q3PPK/R7_b_-_-_0_29?color=white&lastMove=f3f4&variant=standard

Reply:
So maybe it's just a problem with position gifs, in my recent attempts it seems okay for games.

Reply:
Ah, good find! It's not the CDN doing magical things, just bug(s): #11479

Reply:
Nice, thanks for that fix! Ok, I think your strategy could work then. A bit busy at the moment, but hopefully can make the changes soon. I'll go back to draft for now.

Reply:
Just started a discussion in #lila-development of what this alternative might look like, but it's turning out a bit convoluted 🤔